akka-remote/src/main/scala/akka/remote/artery/ArteryTransport.scala

```diff
@@ -379,10 +428,24 @@ private[remote] class ArteryTransport(_system: ExtendedActorSystem, _provider: R
     16,
     create = () ⇒ new ReusableInboundEnvelope, clear = inEnvelope ⇒ inEnvelope.asInstanceOf[ReusableInboundEnvelope].clear())
 
-  val (afrFileChannel, afrFlie, flightRecorder) = initializeFlightRecorder()
+  val (afrFileChannel, afrFlie, flightRecorder) = initializeFlightRecorder() match {
+    case None            ⇒ (None, None, None)
+    case Some((c, f, r)) ⇒ (Some(c), Some(f), Some(r))
+  }
+
+  def createFlightRecorderEventSink(synchr: Boolean = false): EventSink = {
+    flightRecorder match {
+      case Some(f) ⇒
+        val eventSink = f.createEventSink()
+        if (synchr) new SynchronizedEventSink(eventSink)
+        else eventSink
+      case None ⇒
+        IgnoreEventSink
+    }
```

Comment:
@drewhk I tried to to synchronize the ArteryTransport.topLevelFREvents, but still jvm is still crashing. Tried a few other things, but couldn't figure out what is wrong. I have added config to disable flight recorder and it's currently disabled because of this issue.